app/web/src/components/Actions/ActionCard.vue

```diff
@@ -13,43 +13,49 @@
       )
     "
   >
-    <Icon
-      v-if="actionQueued"
-      name="nested-arrow-right"
-      :class="clsx(themeClasses('text-neutral-600', 'text-neutral-300'))"
-    />
-    <Icon
-      v-else-if="actionRunning"
-      name="loader"
-      :class="clsx(themeClasses('text-action-300', 'text-action-300'))"
-    />
-    <Icon
-      v-else-if="actionOnHold"
-      name="circle-stop"
-      :class="
-        clsx(
-          holdStatusInfluencedBy.length > 0
-            ? [
-                'opacity-30',
-                themeClasses('text-warning-500', 'text-warning-300'),
-              ]
-            : themeClasses('text-warning-400', 'text-warning-300'),
-        )
-      "
-      size="lg"
-    />
-    <template v-else-if="actionFailed">
+    <template v-if="actionProposed">
+      <Icon
+        v-if="actionQueued"
+        name="nested-arrow-right"
+        :class="clsx(themeClasses('text-neutral-600', 'text-neutral-300'))"
+      />
       <Icon
-        name="play"
-        :class="clsx(themeClasses('text-action-700', 'text-action-300'))"
-        @click="retry"
+        v-else-if="actionRunning"
+        name="loader"
+        :class="clsx(themeClasses('text-action-300', 'text-action-300'))"
       />
       <Icon
-        name="x-hex-outline"
+        v-else-if="actionOnHold"
+        name="circle-stop"
         :class="
-          clsx(themeClasses('text-destructive-700', 'text-destructive-700'))
+          clsx(
+            holdStatusInfluencedBy.length > 0
+              ? [
+                  'opacity-30',
+                  themeClasses('text-warning-500', 'text-warning-300'),
+                ]
+              : themeClasses('text-warning-400', 'text-warning-300'),
+          )
         "
+        size="lg"
       />
+      <template v-else-if="actionFailed">
+        <Icon
```

Comment:
We were gonna only show this on "proposed", not "history", right? So history can never have 3 icons...

Reply:
That's all within a `<template>` from lines 16-55 that is only for `actionProposed`

I do have a "Failed" result as part of the `ActionHistoryResult` but we can remove it if we end up not using it.